classes/PrestaShopAutoload.php

```diff
@@ -58,7 +65,8 @@ class PrestaShopAutoload
     protected function __construct()
     {
         $this->root_dir = _PS_CORE_DIR_ . '/';
-        $file = static::getCacheFileIndex();
+        $this->classLoader = new LegacyClassLoader(_PS_ROOT_DIR_, _PS_CACHE_DIR_);
```

Comment:
I can't comment on the class name but this is a generic comment related with this class. Since you exported your code in an external library if we follow the idea shouldn't this class be removed as well and replaced with the one from the library?
https://github.com/FabienPapet/PrestaShop/blob/a0d7fb0df20149e8c71c0b982e6dbd255f830333/classes/PrestaShopAutoload.php#L34

Reply:
It can, but I did not do that because removing code is mainly on major versions.

Reply:
That makes sense! Then maybe we should add a deprecation comment indicating that should be replaced by the class from the library in 9.0 ?

Reply:
Deprecate was planned for a future PR, but I can do it if needed